src/components/search/price-filter.js

```diff
@@ -35,76 +34,78 @@ class PriceFilter extends Component {
   }
 
   // Called by filter-group
-  getFilters() {
+  async getFilters() {
     return [
       {
         name: this.props.filter.searchParameterName,
-        value: this.state.value[0],
+        value: await getEthPrice(this.state.value[0], 'USD', 'ETH'),
         valueType: VALUE_TYPE_FLOAT,
         operator: FILTER_OPERATOR_GREATER_OR_EQUAL
       },
       {
         name: this.props.filter.searchParameterName,
-        value: this.state.value[1],
+        value: await getEthPrice(this.state.value[1], 'USD', 'ETH'),
         valueType: VALUE_TYPE_FLOAT,
         operator: FILTER_OPERATOR_LESSER_OR_EQUAL
       }
     ]
   }
 
   componentDidUpdate(previousProps) {
-    // When new search is triggered, search filters get reset, so component should reset their state
-    if (
-      Object.keys(previousProps.filters).length !== 0 &&
-      Object.keys(this.props.filters).length === 0
-    )
+    // New max price property... reset filter
+    if (previousProps.maxPrice !== this.props.maxPrice)
       this.onClear()
   }
 
   // Called by filter-group
-  onClear() {
-    this.handlePriceChange([this.defaultMinimum, this.defaultMaximum])
+  onClear(callback) {
+    this.setState({ value: [
+      Math.round(parseFloat(this.props.minPrice)),
```

Comment:
I would use Math.floor for minPrice and Math.ceil for maxPrice.

Reply:
Ok, changing